doc/rds-day-2.md

```diff
@@ -0,0 +1,98 @@
+
+One of most notable problems is that updates made to certain fields could trigger database restarts. Developers may not know which fields would cause restarts because they are not familiar with underlying technologies. You could document potentially dangerous fields, but it is not enough to reliably stop it from happening. 
+
+## Solutions
+
+### Check during PR
+Use Pull Request as a checkpoint and ensure developers are aware of potential consequences of the changes. An example process may look something like the following. 
+
+
+```mermaid
+flowchart LR
+    FieldDefinitions(Fields that need restarting)
+    StepFiles(Get changed files)
+    StepCheck(Check if fields that require restart changed)
+    Comment(Comment on PR)
+    FieldDefinitions <--reference--> StepCheck
+
+    PR(PR Created) --> CR(Does this need DB Restart) --need restart--> StepFiles --> StepCheck --> Comment --> Approval --> Merge
+    CR --no restart--> Approval
+```
+
+In this example, whenever a pull request is created, a workflow is executed and a comment is created on the PR warning the developers of potential impacts. When developers approve the PR, it implies that they are aware of consequences.
+To check if a PR is impacted, you can use of the following options:
+- Parse git diff and search for changes to "dangerous" fields
+- Use `kubectl diff` then look for changes to "dangerous" fields. This requires read access to the target cluster but more accurate.
+
+In this approach, it is important for the check mechanisms to work reliably. It's easy to lose developers' trust when checks say there will be restarts but no restart happened. Or worse, checks did not detect potential restarts and caused an outage.
+
+### Check at runtime
+
+Another approach is to deny such operation at runtime using a policy engine and/or custom validating web hook unless certain conditions are met. This means problems with RDS configuration is communicated to the developers through their GitOps tooling by providing reasons for denial.
+
+```mermaid
+flowchart LR
+    subgraph Kubernetes
+        ConfigMap(ConfigMap w/ ticket numbers)
+        ValidatingController(Policy Engine / Validating controller)
+    end 
+
+    subgraph Git 
+        PR(PR Merged)
+    end
+
+    subgraph Ticketing
+       Approved(Approved Changes)
+    end
+    
+    GitOps(GitOps tooling)
+
+    PR(PR Merged) --> GitOps --> ValidatingController
+    ValidatingController --reference--> ConfigMap
+    ValidatingController --deny and provide reason--> GitOps
+    Approved --push--> ConfigMap
+```
+In the example above, no check is performed during PR. During admission into the Kubernetes cluster, a validating controller will lookup config map which contain ticket number and validate the request is valid. If no ticket number associated with this change is approved, it's rejected with provided reason. 
```

Comment:
I would add to this solution an example of "and validate the request is valid." to have the ticketing system do 2 things, 1. create the configmap with unique ticket number, and 2. go back to git and annotate the claim with the ticket number. next time gitops try to reconcile the validation controller checks if the ticketnumber on the claim has a corresponding configmap if it does it allows, if no configmap it denies.

Reply:
Thanks! Added this to the doc.